packages/clerk-js/bundle-check.mjs

```diff
@@ -0,0 +1,238 @@
+            }
+
+            push(path) {
+                const newUrl = new URL(this.#url.toString());
+                newUrl.pathname = path;
+
+                this.#url = newUrl;
+                this.emit();
+            }
+
+            replace(path) {
+                this.push(path);
+            }
+
+            shallowPush(path) {
+                this.push(path);
+            }
+
+            pathname() {
+                return this.#url.pathname;
+            }
+
+            searchParams() {
+                return this.#url.searchParams;
+            }
+
+            subscribe(listener) {
+                this.#listeners.add(listener);
+
+                return () => this.#listeners.delete(listener);
+            }
+
+            emit() {
+                this.#listeners.forEach(listener => listener(this.#url));
+            }
+
+            getSnapshot() {
+                return this.#url;
+            }
+        }
+        window.VIRTUAL_ROUTER = new VirtualRouter('/');
+        </script>
+        <script type="text/javascript">${script}</script>
+    </body>
+</html>
+`;
+}
+
+function signIn() {
+  const script = `
+window.Clerk.load({ router: window.VIRTUAL_ROUTER }).then(() => {
+  window.Clerk.mountSignIn(document.getElementById("app"), {});
+});
+`;
+  return template(script);
+}
+
+function signUp() {
+  const script = `
+window.Clerk.load({ router: window.VIRTUAL_ROUTER }).then(() => {
+  window.Clerk.mountSignUp(document.getElementById("app"), {});
+});
+`;
+  return template(script);
+}
+
+/**
+ * Map of URL routes to functions that return the HTML content of the page. This script will generate a report for each
+ * of the routes defined here.
+ */
+const routes = {
+  '/sign-in': signIn(),
+  '/sign-up': signUp(),
+};
+
+const server = http
+  .createServer((req, res) => {
+    const onError = err => {
+      if (err) {
+        res.end();
+        console.error(err);
+      }
+    };
+
+    if (req.url && req.url in routes) {
+      res.writeHead(200, { 'content-type': 'text/html' });
+      res.end(routes[req.url]);
+    } else {
+      const filePath = `./dist${req.url}`;
+      const extname = path.extname(filePath);
+      if (fs.existsSync(filePath) && (extname === '.js' || extname === '.css')) {
+        const contentType = extname === '.js' ? 'text/javascript' : 'text/css';
+        res.writeHead(200, { 'content-encoding': 'gzip', 'content-type': contentType, vary: 'Accept-Encoding' });
+        // We specifically use gzip here since that's the bundle size we really care about.
+        pipeline(fs.createReadStream(filePath), zlib.createGzip(), res, onError);
+      } else {
+        res.writeHead(404, { 'content-type': 'text/plain' });
+        res.end('404 Not Found\n');
+      }
+    }
+  })
+  .listen(4000);
+
+const byteFormatter = Intl.NumberFormat('en', {
+  notation: 'compact',
+  style: 'unit',
+  unit: 'byte',
+  unitDisplay: 'narrow',
+});
+/**
+ * Format bytes into kilobytes
```

Comment:
`Intl.NumberFormat` with `notation: 'compact'` and `unit: 'byte'` choose the most appropriate unit and scale the number accordingly, so not necessarily kilobytes:


```js
const formatter = Intl.NumberFormat("en", {
  notation: "compact",
  style: "unit",
  unit: "byte",
  unitDisplay: "narrow",
});

console.log(formatter.format(10)); // 10B
console.log(formatter.format(200000)); // 200KB
console.log(formatter.format(50000000)); // 50MB
```

```suggestion
 * Format bytes into a human-readable string with appropriate units
```

Reply:
ooo great catch!